File: scripts/mcp_bridge.js
#!/usr/bin/env node
/**
 * MCP Bridge — Node.js wrapper that spawns the C++ MCP server
 * and relays stdin/stdout with protocol format matching.
 */
const { spawn } = require('child_process');

const SERVER = '/home/you_dian/MCP/mcp_mt/build/server';
const CWD = '/home/you_dian/MCP/mcp_mt';

// Log to file, not stderr
const fs = require('fs');
const log = (msg) => fs.appendFileSync('/home/you_dian/MCP/logs/mcp_bridge.log', `[${new Date().toISOString()}] ${msg}\n`);

log(`Bridge starting: ${SERVER}`);

const proc = spawn(SERVER, ['--no-http'], { cwd: CWD, stdio: ['pipe', 'pipe', 'pipe'] });

proc.on('error', (err) => {
    log(`Spawn error: ${err.message}`);
    // Report error to Claude Code via stdout (MCP error format)
    process.stdout.write(JSON.stringify({
        jsonrpc: "2.0", id: 0,
        error: { code: -32000, message: `Spawn failed: ${err.message}` }
    }) + '\n');
    process.exit(1);
});

proc.on('spawn', () => {
    log(`Server spawned, PID=${proc.pid}`);
});

// Relay stdin → server
process.stdin.on('data', (data) => {
    log(`CLIENT->SERVER (${data.length} bytes)`);
    proc.stdin.write(data);
});

process.stdin.on('end', () => {
    log('Client stdin closed');
    proc.stdin.end();
});

// Relay server stdout → client
proc.stdout.on('data', (data) => {
    log(`SERVER->CLIENT (${data.length} bytes)`);
    process.stdout.write(data);
});

// Relay server stderr → file
proc.stderr.on('data', (data) => {
    fs.appendFileSync('/home/you_dian/MCP/logs/mcp_bridge_stderr.log', data);
});

proc.on('close', (code) => {
    log(`Server exited with code ${code}`);
    process.exit(code || 0);
});

process.on('SIGTERM', () => { proc.kill(); process.exit(); });
process.on('SIGINT', () => { proc.kill(); process.exit(); });

File: scripts/mcp_bridge.py
#!/usr/bin/env python3
"""MCP Bridge — relays stdin/stdout to C++ MCP server via pipe."""
import subprocess, sys, os, threading

SERVER = '/home/you_dian/MCP/mcp_mt/build/server'
CWD = '/home/you_dian/MCP/mcp_mt'

proc = subprocess.Popen(
    [SERVER, '--no-http'],
    stdin=subprocess.PIPE, stdout=subprocess.PIPE,
    stderr=subprocess.PIPE, cwd=CWD
)

# Relay stderr to file
def relay_stderr():
    with open('/home/you_dian/MCP/logs/mcp_bridge_stderr.log', 'ab') as f:
        for data in iter(proc.stderr.read, b''):
            f.write(data)

threading.Thread(target=relay_stderr, daemon=True).start()

# Relay stdin → server (in background)
def relay_stdin():
    try:
        while True:
            data = sys.stdin.buffer.read(65536)
            if not data:
                proc.stdin.close()
                break
            proc.stdin.write(data)
            proc.stdin.flush()
    except Exception:
        pass

threading.Thread(target=relay_stdin, daemon=True).start()

# Relay server stdout → client (main thread)
try:
    while True:
        data = proc.stdout.read(65536)
        if not data:
            break
        sys.stdout.buffer.write(data)
        sys.stdout.buffer.flush()
except Exception:
    pass
finally:
    proc.terminate()
    proc.wait()
